File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-9.cs
public class Solution {
    public int LargestRectangleArea(int[] heights) {
        int n = heights.Length;
        int ans = 0;
        for (int i = 0 ; i < n; ++i) {
            int currentMin = heights[i];
            int currentAns = 0;
            int currentWidth = 0;
            for (int j = i; j < n; j++) {
                currentMin = Math.Min(currentMin, heights[j]);
                currentWidth++;
                currentAns = Math.Max(currentAns, currentMin * currentWidth);
            }
            ans = Math.Max(ans, currentAns);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-9.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0, r = 0;
        int ans = 0;
        int n = s.size();

        unordered_map<char,bool> visited;
        while(r < n) {
            if(visited[s[r]] == false) {
                visited[s[r]] = true;
                r++;
            }
            else {
                visited[s[l]] = false;
                l++;
            }
            ans = max(ans, r - l);
        }
        return ans;
    }
};
